src/generator.js

```diff
@@ -34,7 +33,10 @@ export class RoughGenerator {
 
   get lib() {
     if (!this._renderer) {
-      if (self && self.workly && this.config.async && (!this.config.noWorker)) {
+      const hasSelf = typeof self !== 'undefined';
+      
+      if (hasSelf && self.workly && this.config.async && (!this.config.noWorker)) {
+        self._roughScript = self.document && self.document.currentScript && self.document.currentScript.src;
         const tos = Function.prototype.toString;
```

Comment:
We cannot call `self.document.currentScript` inside the get lib method. It has to be used in the outermost scope to get the correct path. 
This implementation is a bit messy anyway. I'll refactor this for a better implementation.